### ISSUE TITLE
Support for openssl-1.1.0f

### ISSUE THREAD

**Issue by Preston-Landers** [pyca/cryptography#3605]
Cryptography fails to compile when linked against openssl-1.1.0f (released 25 May 2017). It works with previous version openssl-1.1.0e.

Platform: Linux - CentOS 7 - x86-64
Python: 3.6.1

Here's the compile output from pip install on the source package. There's a bunch of warnings after that but I am omitting them for brevity. I can provide those if need be.

```
Installing collected packages: cryptography
  Running setup.py install for cryptography: started
    Running setup.py install for cryptography: finished with status 'error'
    Complete output from command /home/planders/s/newpython/dist/pd/Linux/python/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-qt29snub/cryptography/setup.py';f=getattr(tokenize, 'open'
, open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" build_ext -L/home/planders/s/newpython/dist/pd/Linux/openssl/lib -I/home/planders/s/newpython/dist/pd/Linux/
openssl/include install --record /tmp/pip-4dhdbk9y-record/install-record.txt --single-version-externally-managed --compile:
    running build_ext
    generating cffi module 'build/temp.linux-x86_64-3.6/_padding.c'
    creating build
    creating build/temp.linux-x86_64-3.6
    generating cffi module 'build/temp.linux-x86_64-3.6/_constant_time.c'
    generating cffi module 'build/temp.linux-x86_64-3.6/_openssl.c'
    building '_openssl' extension
    creating build/temp.linux-x86_64-3.6/build
    creating build/temp.linux-x86_64-3.6/build/temp.linux-x86_64-3.6
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/planders/s/newpython/dist/pd/Linux/openssl/include -I/home/planders/s/newpython/dist/pd/Linux/py
thon/include/python3.6m -c build/temp.linux-x86_64-3.6/_openssl.c -o build/temp.linux-x86_64-3.6/build/temp.linux-x86_64-3.6/_openssl.o
    build/temp.linux-x86_64-3.6/_openssl.c:756:23: error: conflicting types for ‘ASN1_TIME_to_generalizedtime’
     ASN1_GENERALIZEDTIME *ASN1_TIME_to_generalizedtime(ASN1_TIME *,
                           ^
    In file included from build/temp.linux-x86_64-3.6/_openssl.c:468:0:
    /home/planders/s/newpython/dist/pd/Linux/openssl/include/openssl/asn1.h:628:23: note: previous declaration of ‘ASN1_TIME_to_generalizedtime’ was here
     ASN1_GENERALIZEDTIME *ASN1_TIME_to_generalizedtime(const ASN1_TIME *t,
                           ^
```

**Comment by reaperhulk** [comment]
This is fixed in #3603 and will be included in 1.9. Thanks for the report though!

**Comment by Preston-Landers** [comment]
Sorry I missed that one. Thanks!
